paf-mvp-core-js/src/express/errors.ts

```diff
@@ -0,0 +1,12 @@
+export class UnableToIdentifySignerError extends Error {
```

Comment:
You mean implement ? I don't think so for two reasons:

1. We should somehow separate internal server errors and public errors returned in the http response. Maybe we should create an `internalServerError` class ? 
2. This error is expected to be thrown by the publicKeyProvider. If I understand well, the publicKeyProvider is to be provided by the publisher ? For now, we are using our own implementation but I think that we need to define a stronger interface rather than just `interface PublicKeyProvider {
  (domain: string): Promise<PublicKey>;
}`. We should specify expected errors and it will not be a **NodeError**.

Reply:
You mean that this error is only internal and it should be caught before generating a proper error at the handler level, right? If so, I have been tricked by the location of the file (`src/express/` path). Intuitively, it means to me that it can be used by express handlers. So maybe it is not the right location, isn't it?

Concerning your question, I am not sure that I understand what you mean "by provided by the publisher". I don't think it should be provided by the publisher. However, the implementation of this interface should fetch the "identity" endpoint of the publisher (or any participant that signed something) for getting the public keys. So considering the interface, I think that you have been tricked by the name ("PublicKeyProvider"). This name hides all the logics concerning the identity endpoints.

Reply:
You mean that this error is only internal and it should be caught before generating a proper error at the handler level, right? If so, I have been tricked by the location of the file (`src/express/` path). Intuitively, it means to me that it can be used by express handlers. So maybe it is not the right location, isn't it?

Concerning your question, I am not sure that I understand what you mean "by provided by the publisher". I don't think it should be provided by the publisher. However, the implementation of this interface should fetch the "identity" endpoint of the publisher (or any participant that signed something) for getting the public keys. So considering the interface, I think that you have been tricked by the name ("PublicKeyProvider"). This name hides all the logics concerning the identity endpoints.

I would prefer something like that:
```typescript
interface IdentityPublicKey {
    start: Date,
    end: Date,
    pemKey: string,
}

interface Identity {
    dpoEmail: string,
    privacyPolicyUrl: string,
    name: string,
    keys: IdentityPublicKey[],
    getKeyForDate(date: Date): IdentityPublicKey,
}


interface IIdentityService {
    getIdentity(domain: string): Promise<Identity>
}

// Hypothetical usage
const verifier = ...; 
const idAndPrefs = ...;
const signatureStrBuilder = ...;
const identityService: IIdentityService = ...;
const operatorIdentity = identityService.getIdentity(idAndPrefs.source.domain);
const key = operatorIdentity.getKeyForDate(idAndPrefs.source.date);
const signatureStr = signatureStrBuilder.generate(idAndPrefs);
const isValid =  verifier.validate(signatureStr, idAndPrefs.source.signature, key.pemKey);
```

Reply:
> Intuitively, it means to me that it can be used by express handlers 
Well, it should be used by express handlers to generate a proper response (with a proper error). So yes it is to be used by middlewares, but not returned as a response.